src/dialects/mariadb/query.js

```diff
@@ -34,8 +34,6 @@ export class MariaDbQuery extends AbstractQuery {
 
     let results;
 
-    const errForStack = new Error();
```

Comment:
Moving the `errForStack` error to right before `formatError` is called defeats the purpose of this error. 
The goal is to get a stacktrace before we query, to make sure stacktrace includes the method from the user's codebase that caused this call (instead of what we had before https://github.com/sequelize/sequelize/pull/13347)

The creation errForStack must remain where it is